src/node_process_methods.cc

```diff
@@ -471,6 +473,94 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
   env->Exit(code);
 }
 
+#ifdef __POSIX__
+inline int persist_standard_stream(int fd) {
+  int flags = fcntl(fd, F_GETFD, 0);
+
+  if (flags < 0) {
+    return flags;
+  }
+
+  flags &= ~FD_CLOEXEC;
+  return fcntl(fd, F_SETFD, flags);
+}
+
+static void Execve(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kChildProcess, "");
+
+  CHECK(args[0]->IsString());
+  CHECK(args[1]->IsArray());
+  CHECK(args[2]->IsArray());
+
+  Local<Array> argv_array = args[1].As<Array>();
+  Local<Array> envp_array = args[2].As<Array>();
+
+  // Copy arguments and environment
+  Utf8Value executable(isolate, args[0]);
+  std::vector<std::string> argv_strings(argv_array->Length());
+  std::vector<std::string> envp_strings(envp_array->Length());
+  std::vector<char*> argv(argv_array->Length() + 1);
+  std::vector<char*> envp(envp_array->Length() + 1);
+
+  for (unsigned int i = 0; i < argv_array->Length(); i++) {
+    Local<Value> str;
+    if (!argv_array->Get(context, i).ToLocal(&str)) {
+      THROW_ERR_INVALID_ARG_VALUE(env, "Failed to deserialize argument.");
+      return;
+    }
+
+    argv_strings[i] = Utf8Value(isolate, str).ToString();
+    argv[i] = argv_strings[i].data();
+  }
+  argv[argv_array->Length()] = nullptr;
+
+  for (unsigned int i = 0; i < envp_array->Length(); i++) {
+    Local<Value> str;
+    if (!envp_array->Get(context, i).ToLocal(&str)) {
+      THROW_ERR_INVALID_ARG_VALUE(
+          env, "Failed to deserialize environment variable.");
+      return;
+    }
+
+    envp_strings[i] = Utf8Value(isolate, str).ToString();
+    envp[i] = envp_strings[i].data();
+  }
+
+  envp[envp_array->Length()] = nullptr;
+
+  // Set stdin, stdout and stderr to be non-close-on-exec
+  // so that the new process will inherit it.
+  if (persist_standard_stream(0) < 0 || persist_standard_stream(1) < 0 ||
+      persist_standard_stream(2) < 0) {
+    env->ThrowErrnoException(errno, "fcntl");
+  }
+
+  // Perform the execve operation.
+  RunAtExit(env);
+  execve(*executable, argv.data(), envp.data());
+
+  // If it returns, it means that the execve operation failed.
+  // In that case we abort the process.
+  auto error_message = std::string("process.execve failed with error code ") +
+                       errors::errno_string(errno);
+
+  // Abort the process
+  Local<v8::Value> exception =
+      ErrnoException(isolate, errno, "execve", *executable);
+  Local<v8::Message> message = v8::Exception::CreateMessage(isolate, exception);
+
+  std::string info = FormatErrorMessage(
+      isolate, context, error_message.c_str(), message, true);
+  FPrintF(stderr, "%s\n", info);
+  ABORT();
```

Comment:
Why does this abort instead of throwing?

Reply:
@addaleax ... see https://github.com/nodejs/node/pull/56496#discussion_r1912061490

I think this is something we can reasonably debate a bit.

Reply:
Hm yeah ... I'd argue that this further limits the already fairly limited usefulness of this feature, but I understand where it's coming from 🙂 Just seems odd that this hard crashes in this case, you can't always know for sure that `execve()` will succeed

Reply:
Yep, this one is rather tricky. I'm not quite sure what this should do really and absolutely open to alternatives. One suggestion is to mark this new API as experimental, stick with the abort for now, and see if a reasonable alternative emerges before we graduate it but I'm not really certain what the ideal approach here is.

Reply:
Done, the API is now experimental, we can revisit this in the future.